src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -228,6 +228,80 @@ class E
             comp.VerifyDiagnostics();
         }
 
+        [Fact, WorkItem(49263, "https://github.com/dotnet/roslyn/issues/49263")]
+        public void DeriveFromSelf()
+        {
+            var src = @"
+record R : R;
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics(
+                // (2,8): error CS0146: Circular base type dependency involving 'R' and 'R'
+                // record R : R;
+                Diagnostic(ErrorCode.ERR_CircularBase, "R").WithArguments("R", "R").WithLocation(2, 8),
+                // (2,8): error CS0115: 'R.ToString()': no suitable method found to override
+                // record R : R;
+                Diagnostic(ErrorCode.ERR_OverrideNotExpected, "R").WithArguments("R.ToString()").WithLocation(2, 8),
+                // (2,8): error CS0115: 'R.EqualityContract': no suitable method found to override
+                // record R : R;
+                Diagnostic(ErrorCode.ERR_OverrideNotExpected, "R").WithArguments("R.EqualityContract").WithLocation(2, 8),
+                // (2,8): error CS0115: 'R.Equals(object?)': no suitable method found to override
+                // record R : R;
```

Comment:
I can already see this behavior on the current master with SharpLab.

![image](https://user-images.githubusercontent.com/31348972/106379919-e52c9000-63b7-11eb-9c8b-a1cd9ff19e84.png)

Reply:
Removed the duplicate test. Thanks